202506 satellite health

eulaliesa opened the pull request:
This commit refractors the _evaluate.py_ pipeline and its test to directly use RINEX OBS files instead of pre-processed _.prx_ files for generating query data. Queries are now builts from actual RINEX OBS content using the function _parse_rinex_obs_file_ , filtering out any cycle slip indicators, and renaming columns to match the expected structure. Health flags are computed for each satellite observation, and the test checks that all values are valid and complete, with no missing data (_NaN_). It also includes specific cases manually validated to ensure that the output is correct. Finally, unused code was cleaned from _evaluate.py_, in order the make the implementation cleaner and more aligne with real observational input data.

jtec commented on src/prx/test/datasets/TLSE00FRA_R_2024001/TLSE00FRA_R_20240011800_01H_30S_MO.crx.gz:
Could we use one of the existing test files here?

eulaliesa replied:
Indeed, I chose this test file because the satellite G27 had an unhealthy flag. The test was made to check how the position error was influenced by an unhealthy satellite.

jtec commented on src/prx/rinex_nav/evaluate.py:
```diff
@@ -653,6 +685,7 @@ def evaluate_orbit(sub_df):
 
     per_sat_eph_query = per_sat_eph_query.groupby("orbit_type").apply(evaluate_orbit)
     per_sat_eph_query = per_sat_eph_query.reset_index(drop=True)
+    per_sat_eph_query["health_flag"] = extract_health_flag_from_query(per_sat_eph_query)
```
I recommend a shorter

```
query["health_flag"] = query["health"]
query.loc[query.constellation=="C", "health_flag"] =  query.loc[query.constellation=="C", "SatH1"] 
```

Vectorized pandas operations tend to be (much) faster than looping over rows :).

jtec commented:
@eulaliesa I am not sure I understand "directly use RINEX OBS files instead of pre-processed .prx files for generating query data." in the PR description - the CSV file (what does ".prx files" refer to?) is generated in the very end, so the query is always generated from observation timestamps. What am I missing?

eulaliesa commented:
@jtec I modified `evaluate.py` and `test_evaluate.py` based on your comments. I deleted `test_extract_health_flag_from_query` and changed the function `extract_health_flag_from_query` to use vectorized pandas operations. As for my comment, I am sorry that wasn't clear and I missunderstood. The query is indeed generated from observation timestamps.

jtec commented:
@eulaliesa Thanks for addressing the comments :) I took the liberty to update the PR description with the wording suggested by @plutonheaven. We try to keep the PR descriptions as clear as possible, since they are typically what future developers will look at to better understand a particula commit. PR descriptions are versioned, so feel free to roll back the changes I made if you disagree :)  

Did you have a chance to run https://github.com/jtec/prx/blob/aaef97322f0f7d118947c203f2c6deaf72695eb0/src/prx/test/benchmark.py#L36 as a quick check of how run time is impacted by this PR?

Running those benchmarks automatically in CI has been on our list for a while but did not make it into the code yet.

eulaliesa commented:
@jtec I ran `benchmark.py` on both the master branch and the updated one. The new version was approximately 4.2 secondes faster. Let me know if you think some changes still need to be done. :)
